libs/@blockprotocol/core/src/service-handler.ts

```diff
@@ -48,27 +63,56 @@ export abstract class ServiceHandler {
     serviceName,
     sourceType,
   }: {
-    element: HTMLElement;
+    element?: HTMLElement | null;
     serviceName: string;
     sourceType: "block" | "embedder";
   }) {
-    this.element = element ?? null;
     this.serviceName = serviceName;
     this.sourceType = sourceType;
 
+    if (element) {
+      this.registerService(element);
+    }
+  }
+
+  initialize(element: HTMLElement) {
+    if (!this.element) {
+      this.registerService(element);
+    }
+
+    const coreHandler = this.coreHandler;
+
+    if (!coreHandler) {
+      throw new Error("Could not initialize – missing core handler");
+    }
+
+    coreHandler.initialize();
+
+    this.processCoreCallbacks();
+  }
+
+  registerService(element: HTMLElement) {
```

Comment:
I think this can be private
```suggestion
  private registerService(element: HTMLElement) {
```